numpy/core/src/multiarray/datetime_busdaycal.c

```diff
@@ -75,7 +73,7 @@ PyArray_WeekMaskConverter(PyObject *weekmask_in, npy_bool *weekmask)
 general_weekmask_string:
         /* a string like "SatSun" or "Mon Tue Wed" */
         memset(weekmask, 0, 7);
-        for (i = 0; i < len; i += 3) {
+        for (int i = 0; i < len; i += 3) {
```

Comment:
```suggestion
        for (Py_ssize_t i = 0; i < len; i += 3) {
```
I know it was already the case, but should be `ssize_t`, the rest of the code seems compatible with that, and I expect some compilers might even warn about it?

Reply:
And without it, this code blows up on strings longer than 2**32 characters